__snapshots__/make-badge.spec.js

```diff
@@ -1,5 +1,5 @@
 exports['The badge generator SVG should always produce the same SVG (unless we have changed something!) 1'] = `
-<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="88" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="88" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h45v20H0z"/><path fill="#4c1" d="M45 0h43v20H45z"/><path fill="url(#b)" d="M0 0h88v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"><text x="235" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="350">cactus</text><text x="235" y="140" transform="scale(.1)" textLength="350">cactus</text><text x="655" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="330">grown</text><text x="655" y="140" transform="scale(.1)" textLength="330">grown</text></g> </svg>
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="88" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="88" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h45v20H0z"/><path fill="#4c1" d="M45 0h43v20H45z"/><path fill="url(#b)" d="M0 0h88v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"> <text x="235" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="350">cactus</text><text x="235" y="140" transform="scale(.1)" textLength="350">cactus</text><text x="655" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="330">grown</text><text x="655" y="140" transform="scale(.1)" textLength="330">grown</text></g> </svg>
```

Comment:
Looks like this adds a space. Is that deliberate?

Reply:
Yeah, i couldn't find a way to change the template with the ifs without an extra space being added.

Reply:
If it doesn't affect the display, I'm surprised SVGO doesn't remove it. Okay. I don't think it's a big deal.

In the long run it may be better to move some of this logic from templates into JS. It could help reduce duplication and make these easier to maintain.

Thanks so much for your work on this!

Reply:
The same thing happens on the `_shields_test` style:
template:
```javascript
{
{{?it.colorA}}
  "colorA": {{=JSON.stringify(it.colorA)}},
{{?}}
{{?it.colorB}}
  "colorB": {{=JSON.stringify(it.colorB)}},
{{?}}
  "name": {{=JSON.stringify(it.text[0])}},
  "value": {{=JSON.stringify(it.text[1])}}
}
```
Output:
```json
{


  "colorB": "#4c1",

  "name": "build",
  "value": "passing"
}
```
Not sure what is causing it though.

Reply:
Only way i could seem to do it was:
```javascript
{
{{?it.colorA}}  "colorA": {{=JSON.stringify(it.colorA)}},{{?}}
{{?it.colorB}}  "colorB": {{=JSON.stringify(it.colorB)}},{{?}}
  "name": {{=JSON.stringify(it.text[0])}},
  "value": {{=JSON.stringify(it.text[1])}}
}
```

**Edit:**
Looks like there is a way to strip whitespace
http://olado.github.io/doT/index.html#samples:
![image](https://user-images.githubusercontent.com/7288322/39278326-a8bbfe46-4946-11e8-9c28-9c51267ca86c.png)
Not sure how this option would affect other aspects of the badges though.